catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,260 @@
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+
+    }
+
+    @Override
+    public Map<String, List<Long>> userMap() {
+        Map<String, List<Long>> dataUsageMap = new HashMap<>();
+        try {
+
+            attributesStore.getAllUsers()
+                    .stream()
+                    .filter(MapUtils::isNotEmpty)
+                    .forEach(stringObjectMap -> {
+                                if (MapUtils.isNotEmpty(stringObjectMap)) {
+                                    dataUsageMap.put((String) stringObjectMap.get(
+                                                    AttributesStore.USER_KEY + TXT_PREFIX),
+                                            Arrays.asList((Long) stringObjectMap.get(
+                                                            AttributesStore.DATA_USAGE_KEY
+                                                                    + LNG_PREFIX),
+                                                    (Long) stringObjectMap.get(
+                                                            AttributesStore.DATA_USAGE_LIMIT_KEY
+                                                                    + LNG_PREFIX)));
+                                }
+                            });
+
+        } catch (PersistenceException e) {
+            LOGGER.debug("Unable to get persistent users from the AttributesStore.", e);
+        }
+        return dataUsageMap;
+    }
+
+    @Override
+    public void updateUserDataLimit(Map<String, Long> userMap) {
+        LOGGER.debug("Updating user data limit : {}", userMap.toString());
+
+        for (Map.Entry<String, Long> entry : userMap.entrySet()) {
+            long dataLimit = entry.getValue();
+            String username = entry.getKey();
+            try {
+                attributesStore.setDataLimit(username, dataLimit);
+            } catch (PersistenceException e) {
+                LOGGER.warn("Unable to update properties for {}.", username, e);
+            }
+        }
+    }
+
+    @Override
+    public void updateCronTime(String cronTime) {
+        this.cronTime = parseCronTime(cronTime);
+        setPersistentCronTime();
+        try {
+            configure();
+        } catch (Exception e) {
+            LOGGER.warn("Unable to update Cron Time.");
+        }
+
+    }
+
+    public String parseCronTime(String cronTime) {
+        String[] time = cronTime.split(":");
+        String result = "0+" + time[1] + "+" + time[0] + "+*+*+?";
+        LOGGER.debug("Setting new cron time : {}.", result);
+        return result;
+    }
+
+    @Override
+    public String cronTime() {
+        String[] cronSplit = cronTime.split("\\+");
+        String time = cronSplit[2] + ":" + cronSplit[1];
+        return time;
+    }
+
+    @Override
+    public void configure() {
+
+        from("quartz://myGroupName/myTimerName?cron=" + cronTime + "&stateful=true").errorHandler(
```

Comment:
parseCronTime is responsible for producing a valid cron time to use in Camel Quartz